test/ng/locationSpec.js

```diff
@@ -663,6 +663,22 @@ describe('$location', function() {
       });
     });
 
+
+    it('should not get caught in infinite digest when replacing path in locationChangeSuccess handler', function() {
+      initService({html5Mode:true,supportHistory:false});
+      mockUpBrowser({initialUrl:'http://server/base/home', baseHref:'/base/'});
+      inject(
+        function($browser, $location, $rootScope, $window) {
+          $rootScope.$on('$locationChangeSuccess', function() {
+            if ($location.path() !== '/') {
+                $location.path('/').replace();
+            }
+          });
+          $rootScope.$digest();
```

Comment:
need to check that the change handler actually gets called and that the path is updated correctly